libcontainer/rootfs_linux.go

```diff
@@ -437,9 +488,13 @@ func mountToRootfs(m *configs.Mount, c *mountConfig) error {
 		if err := prepareBindMount(m, rootfs, mountFd); err != nil {
 			return err
 		}
-		if err := mountPropagate(m, rootfs, mountLabel, mountFd); err != nil {
-			return err
+
+		if mountByMove(m, dest) != nil {
```

Comment:
I'm testing this, and it seems this can't work. At this point we are inside the userns (I've added some `logrus.Debugf` to print the content of `/proc/self/uid_map` to verify this) and therefore we create an idmap mount. They only work from the initial userns.

Did you made this runc code work for you at some point? Or was this a c&p from working code in a standalone binary (that you tried from the host, so the initial userns) that never was tested in the runc binary (that IIUC this part runs inside the userns).

Am I missing something? Or do we need a big rework to make this work?